realm/realm-library/src/main/java/io/realm/internal/Table.java

```diff
@@ -629,6 +655,25 @@ void checkIntValueIsLegal(long columnIndex, long rowToUpdate, long value) {
         }
     }
 
+    // check if it is ok to use null value for given row and column.
+    void checkDuplicatedNullForPrimaryKeyValue(long columnIndex, long rowToUpdate) {
+        if (isPrimaryKeyColumn(columnIndex)) {
+            RealmFieldType type = getColumnType(columnIndex);
+            switch (type) {
+                case STRING:
+                case INTEGER:
+                    long rowIndex = findFirstNull(columnIndex);
+                    if (rowIndex != rowToUpdate && rowIndex != TableOrView.NO_MATCH) {
+                        throwDuplicatePrimaryKeyException("null");
```

Comment:
[`copyToRealm_duplicatedNullPrimaryKeyThrows()`](https://github.com/realm/realm-java/pull/2515/files#diff-eb98b3546f9b692ff01aaff903a3bdb3R1358) test has been created and passes for all nullable primary key classes.